src/command_modules/azure-cli-interactive/azclishell/app.py

```diff
@@ -498,36 +498,33 @@ def handle_jmespath_query(self, args):
         continue_flag = False
 
         if hasattr(self.last.result, '__dict__'):
-            input_dict = dict(self.last.result)
-        else:
-            input_dict = self.last.result
+            input_last = dict(self.last.result)
+        else:  # list or string
+            input_last = self.last.result
```

Comment:
This can be further shortened to
`input_last = dict(self.last.result) if hasattr(self.last.result, '__dict__') else self.last.result`

I do have a question though, you may want to do some research. My understand here is that `hasattr(self.last.result, '__dict__')` is used to determine if the `last.result` is a map like object. Is it the common approach though? `hasattr` isn't a really elegant approach to determine instance type.

Reply:
I looked at the code and it seems like if the last command returned a json object, its already a dictionary. Don't know why this was in the code, so I removed it. Also, looked into it and it seems like a lot of people recommend avoiding `hasattr` in general and using try-excepts and catching the AttributeError.